docs(check_user_name): implement description of exercise

eslopfer opened the pull request.

mikysett commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,63 @@
+
+### Usage
+
+Here is a program to test your function
```
Here unfortunately there is not a unique and standard way this sentence is ended in the rest of the exercises.
Still for new ones and when I modify the old ones I try to enforce two dots at the of the sence ( "... function:") which seems to me the most correct way to end it.

eslopfer replied:
I agree. This one slipped through when using the old subject as template. Thanks.

eslopfer replied:
Done.

mikysett commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,63 @@
+  let user0 = User::new("Didier".to_string(), AccessLevel::Admin);
+  println!("{:?}", check_user_name(&user0));
+
+  let user2 = User::new("Mary".to_string(), AccessLevel::Normal);
```
we pass from `user0` to `user2`, maybe it's better to rename `user0` to `user1` or the two others to 1 and2

eslopfer replied:
Oops, good catch, I deleted an irrelevant example and didn't change the number :slightly_smiling_face: .

eslopfer replied:
Done.

mikysett reviewed (changes requested):
minor style issues and one function to be placed into another paragraph

eslopfer commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,63 @@
+
+For this exercise you will have to create a tool that manages users' access levels.
+
+You will have to create an `AccessLevel` enum wich could be `Guest`, `Normal`, `Admin`.
```
Thanks. Done.

eslopfer commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,63 @@
+
+You will also have to create a `User` struct which has:
+
+- elements:
```
Thanks I renamed it to `Fields`, is that what you meant?

eslopfer commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,63 @@
+  - `acessLevel`: `enum`
+- associated functions:
+  - `new`: which initializes the struct.
+  - `send_name`: which takes only **self** as argument and returns an **Option<&str>** with `None` if the user is a guest or the `name` if the `AccessLevel` has any of the other options.
```
I'll change it. I thought it was the agreed way of doing it, I was basing it on the way it was done in the previous exercise.

eslopfer replied:
Done.

eslopfer commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,63 @@
+- associated functions:
+  - `new`: which initializes the struct.
+  - `send_name`: which takes only **self** as argument and returns an **Option<&str>** with `None` if the user is a guest or the `name` if the `AccessLevel` has any of the other options.
+  - `check_user_name`: which takes a `User`, calls `send_name` and returns a **tuple** with `true` and the user `name` if `send_name` returns the name or `false` and `"Error: User is guest"` if not
```
By reading your proposal I understood you wanted it as an associated function. After reading the link, I still don't see why you say it shouldn't be an associated function since it's related with the struct and you won't use it without it. 
I'll change it anyway.

eslopfer replied:
Done.

nprimo commented on subjects/check_user_name/README.md:
```diff
@@ -0,0 +1,64 @@
+  - `acessLevel`: `enum`
+- Associated functions:
+  - `new`: which initializes the struct.
+  - `send_name`: which takes only `self` as argument and returns an `Option<&str>` with `None` if the user is a guest or the `name` if the `AccessLevel` has any of the other options.
```
I would highlight the word `Guest` to make it even clearer

eslopfer replied:
Thanks. Done

nprimo commented:
I have added a comment for a minor improvement; in any case, it looks good to be merged to me :)

eslopfer commented:
> I have added a comment for a minor improvement; in any case, it looks good to be merged to me :)

I fixed it, also renamed the folder to keep the name convention as Dav instructed.

nprimo reviewed (approved):
LGTM! 🚀
